src/main/java/org/prebid/server/bidder/ogury/OguryBidder.java

```diff
@@ -0,0 +1,285 @@
+            errors.add(BidderError.badInput("There are no valid impressions to create bid request to ogury bidder"));
+            return Result.withErrors(errors);
+        }
+
+        final List<Imp> impsWithOguryParams = new ArrayList<>();
+        List<Imp> imps = modifyImps(request, impsWithOguryParams::add);
+
+        final BidderError error = validateRequestKeys(request, impsWithOguryParams);
+        if (error != null) {
+            errors.add(error);
+            return Result.withErrors(errors);
+        }
+
+        if (CollectionUtils.isNotEmpty(impsWithOguryParams)) {
+            imps = impsWithOguryParams;
+        }
+
+        final BidRequest modifiedBidRequest = request.toBuilder().imp(imps).build();
+        final MultiMap headers = buildHeaders(modifiedBidRequest);
+        httpRequests.add(BidderUtil.defaultRequest(modifiedBidRequest, headers, endpointUrl, mapper));
+
+        return Result.of(httpRequests, errors);
+    }
+
+    @Override
+    public Result<List<BidderBid>> makeBids(BidderCall<BidRequest> httpCall, BidRequest bidRequest) {
+        final HttpResponse response = getResponse(httpCall);
+        if (response == null || isNotHasContent(response)) {
+            return Result.empty();
+        }
+
+        final BidderError error = checkResponseStatusCodeForErrors(response);
+        if (error != null) {
+            return Result.withError(error);
+        }
+
+        try {
+            final String body = response.getBody();
+            if (StringUtils.isEmpty(body)) {
+                return Result.empty();
+            }
+
+            final BidResponse bidResponse = mapper.decodeValue(body, BidResponse.class);
+
+            final List<BidderError> errors = new ArrayList<>();
+            final List<BidderBid> bidderBids = extractBids(bidResponse, errors::add);
+
+            return Result.of(bidderBids, errors);
+        } catch (Exception e) {
+            return Result.withError(BidderError.badServerResponse(e.getMessage()));
+        }
+    }
+
+    private List<Imp> modifyImps(BidRequest request, Consumer<Imp> impWithOguryParamsHandler) {
+        return Optional.of(request)
+                .map(BidRequest::getImp)
+                .map(sourceImps -> sourceImps.stream().map(imp -> {
+                    final ObjectNode impExt = resolveImpExt(imp);
+                    final ObjectNode impExtBidderHoist = resolveImpExtBidderHoist(impExt);
+
+                    final ObjectNode modifiedImpExt = modifyImpExt(impExt, impExtBidderHoist);
+                    final BigDecimal bidFloor = resolveBidFloor(request, imp);
+                    final Imp modifiedImp = modifyImp(imp, bidFloor, modifiedImpExt);
+
+                    if (hasOguryParams(impExtBidderHoist)) {
+                        impWithOguryParamsHandler.accept(modifiedImp);
+                    }
+
+                    return modifiedImp;
+                }))
+                .map(Stream::toList).orElse(null);
+    }
```

Comment:
Please, see examples how to iterate over imps in other adapters
See https://github.com/prebid/prebid-server-java/pull/3768/files as a starting point